packages/cypress/src/executors/cypress/cypress.impl.ts

```diff
@@ -154,30 +154,37 @@ async function* startDevServer(
     return;
   }
 
-  const { project, target, configuration } = parseTargetString(
-    opts.devServerTarget
-  );
-  const devServerTargetOpts = readTargetOptions(
-    { project, target, configuration },
-    context
-  );
-  const targetSupportsWatchOpt =
-    Object.keys(devServerTargetOpts).includes('watch');
-
-  for await (const output of await runExecutor<{
-    success: boolean;
-    baseUrl?: string;
-  }>(
-    { project, target, configuration },
-    // @NOTE: Do not forward watch option if not supported by the target dev server,
-    // this is relevant for running Cypress against dev server target that does not support this option,
-    // for instance @nguniversal/builders:ssr-dev-server.
-    targetSupportsWatchOpt ? { watch: opts.watch } : {},
-    context
-  )) {
-    if (!output.success && !opts.watch)
-      throw new Error('Could not compile application files');
-    yield opts.baseUrl || (output.baseUrl as string);
+  const devServerTargets = opts.devServerTarget.split(/\s*,\s*/);
+  for (let i = 0; i < devServerTargets.length; i++) {
+    const { project, target, configuration } = parseTargetString(
+      devServerTargets[i]
+    );
+
+    const devServerTargetOpts = readTargetOptions(
+      { project, target, configuration },
+      context
+    );
+
+    const targetSupportsWatchOpt =
+      Object.keys(devServerTargetOpts).includes('watch');
+
+    for await (const output of await runExecutor<{
```

Comment:
I'm 80% sure this won't behave as expected. Last time I tried this approached there is an issue where other executors (i.e. @nrwl/js:node) makes their own child processes and they won't exit. 

You can test this by making an FE and API (@nrwl/nest:app) and hooking them up. running it the first time should work, but the second time the API port will still be in use. Also need to check if using watch and then pressing ctrl+c and then restarting correctly works (i.e. process properly closed down). Please correct me if I'm wrong here, but that's been my issue with this feature in the past.

Also

Reply:
So far I tested running two angular dev servers and they work really well. This change should technically only add the ability to run more than one server using the same approach as before, meaning if there's a limitation on type of server we support, it's not related to this change but how we launch the server which is existing behavior. I tested watch and ctrl-c before creating this PR and confirmed the behavior is the same as before.

Reply:
Could that issue be related to https://github.com/nrwl/nx/issues/11782?